### PR TITLE
osutil/udev: sync with upstream

### PR DESCRIPTION
**Pull request by stolowski**
Merged upstream changes to go-udev to our "fork" in osutil/udev. The upstream change is the change I implemented to support udev events: https://github.com/pilebones/go-udev/pull/9

I've merged the changes using git subtree - it's magic!
```
git fetch go-udev
git merge -s subtree go-udev/master
```
Marking **simple** as this is just syncing with upstream.

### CODE REVIEW

**Comment by mvo5** [comment]
Unit tests fail on 16.04-32:
```
=== RUN   TestParseUdevEvent
--- FAIL: TestParseUdevEvent (0.00s)
panic: runtime error: slice bounds out of range [recovered]
	panic: runtime error: slice bounds out of range

goroutine 8 [running]:
panic(0x81692a0, 0x18420010)
	/usr/lib/go-1.6/src/runtime/panic.go:481 +0x326
testing.tRunner.func1(0x184c8e40)
	/usr/lib/go-1.6/src/testing/testing.go:467 +0x14f
panic(0x81692a0, 0x18420010)
	/usr/lib/go-1.6/src/runtime/panic.go:443 +0x3fd
github.com/snapcore/snapd/osutil/udev/netlink.parseUdevEvent(0x18414370, 0x42, 0x50, 0x0, 0x0, 0x0)
	github.com/snapcore/snapd/osutil/udev/netlink/_test/_obj_test/uevent.go:133 +0x846
github.com/snapcore/snapd/osutil/udev/netlink.ParseUEvent(0x18414370, 0x42, 0x50, 0x0, 0x0, 0x0)
	github.com/snapcore/snapd/osutil/udev/netlink/_test/_obj_test/uevent.go:178 +0x153
github.com/snapcore/snapd/osutil/udev/netlink.TestParseUdevEvent(0x184c8e40)
	/tmp/static-unit-tests/src/github.com/snapcore/snapd/osutil/udev/netlink/uevent_test.go:178 +0x866
testing.tRunner(0x184c8e40, 0x8236c04)
	/usr/lib/go-1.6/src/testing/testing.go:473 +0x8f
created by testing.RunTests
	/usr/lib/go-1.6/src/testing/testing.go:582 +0x6f2
exit status 2
FAIL	github.com/snapcore/snapd/osutil/udev/netlink	0.005s
```

**Comment by stolowski** [comment]
Uh oh, interesting.. looking into it.

**Comment by stolowski** [comment]
I've fixed the problem which caused panic (it's got merged upstream very quickly as well). The failures are now unrelated to the PR, it's about gcloud quotas.

File: osutil/udev/netlink/conn.go [hunk 1]
```diff
@@ -21,15 +31,15 @@ type UEventConn struct {
...
+func (c *UEventConn) Connect(mode Mode) (err error) {
 
 	if c.Fd, err = syscall.Socket(syscall.AF_NETLINK, syscall.SOCK_RAW, syscall.NETLINK_KOBJECT_UEVENT); err != nil {
 		return
 	}
 
 	c.Addr = syscall.SockaddrNetlink{
 		Family: syscall.AF_NETLINK,
-		Groups: 1, // TODO: demistify this field because msg receive if Groups != 0
+		Groups: uint32(mode),
```
Comment (zyga) [inline osutil/udev/netlink/conn.go]: :heart:

**Pull request merged**